src/metabase/search/filter.clj

```diff
@@ -0,0 +1,112 @@
+(ns metabase.search.filter
+  (:require
+   [honey.sql.helpers :as sql.helpers]
+   [metabase.driver.common.parameters.dates :as params.dates]
+   [metabase.public-settings.premium-features :as premium-features]
+   [metabase.search.in-place.filter :as search.in-place.filter]
+   [metabase.search.spec :as search.spec]
+   [metabase.util :as u]
+   [metabase.util.date-2 :as u.date]
+   [metabase.util.i18n :refer [tru]])
+  (:import
+   (java.time LocalDate)))
+
+(def ^:private context-key->attr
+  {:archived?           :archived
+   :created-at          :created-at
+   :created-by          :creator-id
+   :last-edited-at      :last-edited-at
+   :last-edited-by      :last-editor-id
+   :search-native-query :dataset-query
+   ;; this actually has nothing to do with tables anymore, as we also filter cards.
+   :table-db-id         :database-id
+   :verified            :verified})
```

Comment:
So I wonder what's the reason for that map? Why not store data under the necessary key right away?

Reply:
I went back and forth on this and actually did originally remap from the source column names to names like this originally when populating the index.

We need to get rid of the question mark for archived anyway, table-db-id is kind of legacy name, so it wasn't that clean. I decided I'd rather keep consistency between the index and the source fields where it made sense.

Reply:
Well if it's not hardcoded on the FE - we can get rid of a question mark in the `archived?` anyways, right? If it's hardcoded... then damn that sucks I'd love to fix that :\

Reply:
Hmm I guess I could rename the attrs (losing the ability to use `true` in a few places) rather than adding another mapping on ingestion.

As for renaming the keys in the search context, I don't want to have to propagate that through legacy search. Once legacy search is gone we can put filters with the right names inside a nested :filters map maybe.

Reply:
Oh right changing legacy search is not good right now, I agree. But renaming it twice feels weird, and also disconnect between query string and stored data seems like an argument too...

Reply:
Discussed this and here's a summary:

1. Introduce a "filters" map which maps each key to its type. This will simplify applying constraints and can eventually power the frontend.
2. Since (1) enumerates the filters, having consistent keys means we can drop this map.
3. The context keys are crufty, the attr keys are better and I'd rather update the context.
4. We already translate the API schema (`:search-context.input`) to `SearchContext` type, so we can use that to existing mapping to organize the filters together with consistent name, without FE changes.
5. In future the FE can be more "dumb" and learn about filters from the BE, so they're easy to add and change.
6. We still need to worry about FE hardcoding the names of filters for embedded search components without filter UI, e.g. searching for tables in a DB in query builder.